### PR TITLE
feat(lambda): Add tag propagation support for AWS::Serverless::CapacityProvider

### PR DESCRIPTION
**Pull request by vicheey**
### Issue #, if available
N/A

### Description of changes

This is to support: [Tag propagation - AWS LambdaTag propagation - AWS Lambda](https://docs.aws.amazon.com/lambda/latest/dg/lambda-managed-instances-tag-propagation.html)

Add `ManagedResourceTags` property to `AWS::Serverless::CapacityProvider` for configuring tag propagation to managed EC2/ENI/EBS resources.

**SAM syntax:**
```yaml
ManagedResourceTags:
  Propagate: true|false
  Tags:
    key: value
```

**CFN output (`PropagateTags`):**
- `Propagate: true` → `{Mode: CapacityProvider}`
- `Propagate: false` → `{Mode: None}`
- `Tags: {k: v}` → `{Mode: Explicit, ExplicitTags: [{Key, Value}]}`
- Omitted → nothing emitted (CFN default applies)

**Validation framework enhancement:**
Extends `MUTUALLY_EXCLUSIVE` rules to support `Property.Path=Value` syntax. A property with `=Value` is only considered "present" when its actual value matches. This enables `Propagate=True` + `Tags` → error, while `Propagate: false` + `Tags` → valid (allows Globals deep-merge override pattern).

**Globals support:**
```yaml
Globals:
  CapacityProvider:
    ManagedResourceTags:
      Propagate: true
```
Resource-level override with `{Propagate: false, Tags: {...}}` works correctly via SAM's deep-merge + the new validation rule.

**Intrinsic functions:** `!Ref` and `!Sub` pass through to CFN output in tag values.

### Description of how you validated changes

- `make pr` passed: **4537 tests**, 95.62% coverage
- 19 parametrized unit tests for `=Value` validation syntax (path depths, value types, edge cases)
- 5 parametrized generator unit tests (propagate true/false, explicit tags, empty, integration)
- 4 translator end-to-end tests (success fixture with Globals + 3 partitions, error fixture for mutual exclusivity)
- Intrinsic passthrough verified (`!Ref`, `!Sub` in tag values)
- Tested in fresh venv (Python 3.12, `pip install -e '.[dev]'`)

### Checklist

- [x] Review the [generative AI contribution guidelines](https://github.com/aws/serverless-application-model/blob/develop/CONTRIBUTING.md#ai-usage)
- [x] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [x] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [x] Using correct values
    - [x] Using wrong values
- [x] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by licjun — APPROVED** [review]

File: tests/model/test_validation_rule_value_syntax.py
Comment (licjun) [inline tests/model/test_validation_rule_value_syntax.py:1]: I noticed we have both pytest and unittest styles, not sure if we have a guideline regarding this.

Comment (vicheey) [reply tests/model/test_validation_rule_value_syntax.py:1]: Good catch. No formal guideline exists, but the repo is predominantly TestCase + @parameterized.expand. Updated to match that pattern for consistency. Fixed in the latest commit.

File: samtranslator/internal/schema_source/sam-docs.json [hunk 1]
```diff
@@ -120,6 +120,10 @@
...
       "ExcludedTypes": "A list of EC2 instance types to exclude from the capacity provider.  \n*Type*: List  \n*Required*: No  \n*CloudFormation compatibility*: This property is passed directly to the [`ExcludedInstanceTypes`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-lambda-capacityprovider-instancerequirements.html#cfn-lambda-capacityprovider-instancerequirements-excludedinstancetypes) property of [`InstanceRequirements`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-capacityprovider.html#cfn-lambda-capacityprovider-instancerequirements) of an `AWS::Lambda::CapacityProvider` resource."
     },
+    "sam-property-capacityprovider-managedresourcetags": {
+      "Propagate": "TODO",
```
Comment (seshubaws) [inline samtranslator/internal/schema_source/sam-docs.json:124]: when are we going to fill this out? i see a few more TODOs below too

Comment (vicheey) [reply samtranslator/internal/schema_source/sam-docs.json:124]: Planning to update these once the official SAM documentation lands. Will port the descriptions from there to keep them consistent rather than writing ad-hoc text that might diverge.

**Review by seshubaws — APPROVED** [review]
